// @sidvind/better-ajv-errors/lib/cjs/utils.js
var __defProp = Object.defineProperty;
var __markAsModule = (target) => __defProp(target, "__esModule", { value: true });
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
__export(exports, {
  concatAll: () => concatAll,
  getChildren: () => getChildren,
  getErrors: () => getErrors,
  getSiblings: () => getSiblings,
  isAnyOfError: () => isAnyOfError,
  isEnumError: () => isEnumError,
  isRequiredError: () => isRequiredError,
  notUndefined: () => notUndefined
});
const eq = (x) => (y) => x === y;
const not = (fn) => (x) => !fn(x);
const getValues = (o) => Object.values(o);
const notUndefined = (x) => x !== void 0;
const isXError = (x) => (error) => error.keyword === x;
const isRequiredError = isXError("required");
const isAnyOfError = isXError("anyOf");
const isEnumError = isXError("enum");
const getErrors = (node) => node && node.errors || [];
const getChildren = (node) => node && getValues(node.children) || [];
const getSiblings = (parent) => (node) => getChildren(parent).filter(not(eq(node)));
const concatAll = (xs) => (ys) => ys.reduce((zs, z) => zs.concat(z), xs);
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  concatAll,
  getChildren,
  getErrors,
  getSiblings,
  isAnyOfError,
  isEnumError,
  isRequiredError,
  notUndefined
});
